src/kernels/errors/kernelErrorHandler.ts

```diff
@@ -222,23 +224,35 @@ export abstract class DataScienceErrorHandler implements IDataScienceErrorHandle
 
         // Jupyter kernels, non zmq actually do the dependency install themselves
         if (isCancellationError(err)) {
+            traceVerbose(`Cancelled by user`);
+            this.sendKernelTelemetry(err, errorContext, resource, 'cancelled');
             return KernelInterpreterDependencyResponse.cancel;
         } else if (err instanceof JupyterKernelDependencyError) {
             traceWarning(`Jupyter Kernel Dependency Error, reason=${err.reason}`, err);
-            if (err.reason === KernelInterpreterDependencyResponse.uiHidden) {
+            this.sendKernelTelemetry(err, errorContext, resource, err.category);
+            if (err.reason === KernelInterpreterDependencyResponse.uiHidden && this.kernelDependency) {
                 // At this point we're handling the error, and if the error was initially swallowed due to
                 // auto start (ui hidden), now we need to display the error to the user.
-                const response = this.dependencyManager
-                    ? await this.dependencyManager.installMissingDependencies(err)
-                    : JupyterInterpreterDependencyResponse.cancel;
-                return response === JupyterInterpreterDependencyResponse.ok
-                    ? KernelInterpreterDependencyResponse.ok
-                    : KernelInterpreterDependencyResponse.cancel;
```

Comment:
This was wrong, if ipykernel is missing, then we shouldn't be trying to install jupyter dependencies
Possible users were running into this